paseos/communication/optical_link_model.py

```diff
@@ -0,0 +1,107 @@
+import math
+
+from loguru import logger
+
+from .link_model import LinkModel
+from .optical_receiver_model import OpticalReceiverModel
+from .optical_transmitter_model import OpticalTransmitterModel
+from ..actors.base_actor import BaseActor
+
+
+class OpticalLinkModel(LinkModel):
+    """This class defines an optical link model, containing one transmitter and one receiver."""
+
+    def __init__(
+            self,
+            transmitter_actor: BaseActor,
+            transmitter_device_name: str,
+            receiver_actor: BaseActor,
+            receiver_device_name: str,
+    ) -> None:
+        """Initializes the model.
+
+        Args:
+            transmitter_actor (BaseActor): the transmitter in this link.
+            transmitter_device_name (str): the name of the transmitter device.
+            receiver_actor (BaseActor): the receiver in this link.
+            receiver_device_name (str): the name of the receiver device.
+        """
+        self.wavelength = 1550e-9  # in m
+
+        # Get the transmitter and receiver models from the actor
+        transmitter = transmitter_actor.get_transmitter(transmitter_device_name)
+        receiver = receiver_actor.get_receiver(receiver_device_name)
+
+        super().__init__(
+            transmitter_actor,
+            transmitter,
+            receiver_actor,
+            receiver,
+            frequency=299792458 / self.wavelength,
+        )
+
+        assert isinstance(transmitter, OpticalTransmitterModel), (
+            "An optical transmitter is required " "for this optical link."
+        )
+        assert isinstance(
+            receiver, OpticalReceiverModel
+        ), "An optical receiver is required for this optical link."
+
+        logger.debug("Initializing optical link model.")
+        self.required_BER = 10e-3
```

Comment:
Why 10-3? I would have this value as user-defined value. In addition, this is typically linked with the maximum data-rate that you can have.

Reply:
For optical this the BER for which the equation to determine bitrate is valid. It was quite difficult to find an equation where the bitrate could be calculated from. The one I could find so far is only valid for BER 10E-3 and a wavelength of 1550 nm.